src/analysis/complex/upper_half_plane/basic.lean

```diff
@@ -269,6 +269,22 @@ lemma denom_apply (g : SL(2, ℤ)) (z : ℍ) : denom g z = (↑g : matrix (fin 2
 
 end SL_modular_action
 
+section upper_half_plane_manifold
+
+/--The upper half space as a subset of `ℂ` which is convenient sometimes.-/
+def upper_half_space := {z : ℂ | 0 <  z.im}
```

Comment:
Can you instead make the primary definition `upper_half_plane` (at line 43) to be of type `topological_space.opens ℂ`?  Then we don't need multiple similar objects.

Reply:
I see -- the issue is that if you make `upper_half_plane` an `opens ℂ` rather than a `Type` then you lose the ability to have dot notation with `upper_half_plane`.

So I propose the following instead: Instead of making it a manifold using `opens.charted_space` etc, do so with `open_embedding.singleton_charted_space`, etc.  For example,
```
instance upper_half_plane.charted_space : charted_space ℂ ℍ :=
upper_half_plane.open_embedding_coe.singleton_charted_space 
```
(This can live for now at the start of the modular forms file.). Then `smooth_manifold_with_corners`, etc.  I have to go now but can give a more detailed roadmap if you get stuck.